### ISSUE TITLE
Redirect search traffic to archive.cnx.org URLs to the corresponding pages on CNX

### ISSUE THREAD

**Issue by CourtneyRay** [Connexions/cnx#209]
### Description

**Change** 
Currently, archive URLs are showing up in Google search over 1 million times per month. Other changes are also being made to try to prevent archive from showing up in search at all, but in case these URLs do show up, we want to direct users to the appropriate content on CNX so they get what they need and don't have a bad experience. 

----------------
**Details about Change:** _What, Who, Where and How?_

- **User Type:**
` [ ] ` Admin  ` [ ] ` Content/QA
` [ ] ` Student  ` [ ] ` Teacher
` [ ] ` N/A
- **Username:** 
- **URL:**  
- **Course number:**  
- **Course type:**  

- **Steps**

----------------
**Environment:**

- **Instance:**
` [ ] ` production ` [ ] ` qa ` [ ] ` staging
` [ ] ` dev ` [ ] ` content ` [ ] ` demo
- **OS:**
` [ ] ` Mac OSX ` [ ] ` Windows
` [ ] ` Android ` [ ] ` iOS
- **Browser:**
` [ ] ` Chrome ` [ ] ` Firefox ` [ ] ` Safari
` [ ] ` Opera ` [ ] ` IE ` [ ] ` Edge
- **Browser Version:**  

----------------
**Possible impact on a User:**

This should help prevent users from landing on an archive.cnx.org page and having a bad experience with our content. 

----------------
**Images or Screenshots** (_attach below_)

**Comment by CourtneyRay** [comment]
tangential work: https://app.zenhub.com/workspaces/cnx-team-5af1f4cc12da5e6d74331b60/issues/connexions/cnx/208

**Comment by Bkblodget** [comment]
Redirect archive links to cnx as a quick and temporary solution to prevent google searchers from landing on archive pages (~4500 clicks this month alone). Will still need to fix the canonical tags - this work is captured in the issue shared above by Courtney.

**Comment by CourtneyRay** [comment]
Sadly, actually ~45,000 clicks this month alone @Bkblodget

**Comment by edwoodward** [comment]
The solution for this would involve researching how to write a redirect rule in the Varnish config based on referrer. The rule would

1. Look for a referrer header variable - 
2. Check if the referrer is Google
3. If it is, redirect the user to the same URL on cnx.org. This would mean removing the 'archive.' and the '.html' from the url

**Comment by BryanEliDimas** [comment]
@brittweinstein This one is now in progress :), not 508 on cnx-deploy. I got confused, I think I just talked with Ross about that one but this is the more important one.

**Comment by pumazi** [comment]
I spoke with Bryan about his game plan for this. The psuedo implementation he told me about over a call sounds good (he can type up that psuedo implementation if anyone is interested). He'll be doing the implementation in a branch of `cnx-deploy` and (hopefully) requesting deployment of said branch to qa.cnx.org. The testing strategy will be to put a test in `cnx-automation` and helpfully test this on qa.cnx.org.

@m1yag1, Bryan could use some help finding a good place within `cnx-automation` to put his test as well as guidance on how to write the test. Would this be something you could help him with?

@omehes, Bryan could use some assistance as well as your "ok" to utilize qa.cnx.org for testing the implementation for this. You may also like to be involved in writing the `cnx-automation` test for this.

**Comment by BryanEliDimas** [comment]
Thanks @pumazi! Indeed, at this time I have an idea for what the implementation is going to look like, simply by reverse engineering the VCL logic that is already there. This is what I have in mind atm:

```
if (req.http.host ~ "archive" && req.http.Referer ~ "google" && req.http.url ~ "/contents"){
        return (synth(302, regsub(req.http.url, req.http.Canonical)))
    }
```

@edwoodward and I talked with DevOps and we agreed that they will help me deploy to whatever server is available ✅, one of the dev servers or QA as a last resort.

As for testing, we/I can definitely use `curl` or `wget` at first, and _then_ once the code is working, I plan to write a test in `cnx-automation`.

**Comment by BryanEliDimas** [comment]
@edwoodward,  What should happen if there is no Canonical header variable? I think I could simply remove the `archive` part from the host/url, and that will probably redirect to `cnx.org` even if it's not the canonical version of the book.

**Comment by edwoodward** [comment]
@BryanEliDimas That would be the correct solution.

**Comment by m1yag1** [comment]
> I plan to write a test in cnx-automation but that is not a requirement of this issue before deployment.

Yikes! This sounds a lot like feature delivery is a lot more important than quality. Tests as they are developed can help solve problems. For example, when I created a concourse resource I wrote all of the tests first. Once I had a test in place to make sure what I did was correct the solution was straightforward. My test initially made 3rd party requests and that was ok. Once the test worked I mocked the requests and it became a unit test. The difference here is would not need to be mocked out.

**Comment by BryanEliDimas** [comment]
Good point! Having a (manual) test is definitely a requirement but I don't believe _an automated test_, at this time, is a requirement—according to the discussions I've had with @edwoodward.

**Comment by m1yag1** [comment]
If this is going to become some kind of regression test it needs to be automated. Especially something like this which can be done more easily with code. It can also be triggered by all sorts of events to make sure things continue to work. @edwoodward will this become something @omehes will need to manually test during regression testing? If so, I highly suggest a test gets written as part of the acceptance criteria for this work.

**Comment by edwoodward** [comment]
@m1yag1 We are running into 2 deadlines for this issue. One is we need to get it completed this quarter which ends for coding in a week. The other is the pending vacation of the developer which is scheduled to start on Wednesday. I suggested he get it coded and have a solid manual test and then see if we can get the automation done. If we fall back on the manual test, it would only be for this release. He would complete the automation when he returns from vacation. We are all in favor of automating everything we can.

**Comment by m1yag1** [comment]
@edwoodward I definitely understand about deadlines. I'm also not opposed to pairing and helping build the test along with the implementation.
